maven-embedder/src/main/java/org/apache/maven/cli/internal/BootstrapCoreExtensionManager.java

```diff
@@ -128,19 +128,27 @@ private CoreExtensionEntry createExtension( CoreExtension extension, List<Artifa
 
     private List<Artifact> resolveExtension( CoreExtension extension, RepositorySystemSession repoSession,
                                              List<RemoteRepository> repositories, DependencyFilter dependencyFilter )
-        throws PluginResolutionException
+        throws ExtensionResolutionException
     {
-        Plugin plugin = new Plugin();
-        plugin.setGroupId( extension.getGroupId() );
-        plugin.setArtifactId( extension.getArtifactId() );
-        plugin.setVersion( extension.getVersion() );
-
-        DependencyNode root =
-            pluginDependenciesResolver.resolveCoreExtension( plugin, dependencyFilter, repositories, repoSession );
-        PreorderNodeListGenerator nlg = new PreorderNodeListGenerator();
-        root.accept( nlg );
-        List<Artifact> artifacts = nlg.getArtifacts( false );
-
-        return artifacts;
+        try
+        {
+            Plugin plugin = new Plugin();
+            plugin.setGroupId( extension.getGroupId() );
+            plugin.setArtifactId( extension.getArtifactId() );
+            plugin.setVersion( extension.getVersion() );
+
+            DependencyNode root = pluginDependenciesResolver
```

Comment:
Can you add a TODO that the interface likely needs to change to receive `CoreExtension` directly?

Reply:
Fixed